packages/jest-cli/src/runGlobalHook.js

```diff
@@ -27,24 +33,42 @@ export default ({
   }
 
   if (globalModulePaths.size > 0) {
-    return Promise.all(
-      Array.from(globalModulePaths).map(async modulePath => {
-        if (!modulePath) {
-          return null;
-        }
-
-        // $FlowFixMe
-        const globalModule = require(modulePath);
-
-        if (typeof globalModule !== 'function') {
-          throw new TypeError(
-            `${moduleName} file must export a function at ${modulePath}`,
-          );
-        }
-
-        return globalModule(globalConfig);
-      }),
-    );
+    return pEachSeries(Array.from(globalModulePaths), async modulePath => {
+      if (!modulePath) {
+        return;
+      }
+
+      const projectConfig =
+        allTests
+          .map(t => t.context.config)
+          .find(c => c[moduleName] === modulePath) ||
```

Comment:
I don't see a reason to loop over `allTests` twice, single `find` will be just enough

Reply:
sure